### ISSUE TITLE
Cannot use CombinedEffect inside SequenceEffect

### ISSUE THREAD

**Issue by aliasonym** [flame-engine/flame#463]
## Description

When trying to add a SequenceEffect to a PositionComponent with a CombinedEffect inside it, the following exception is thrown:

```
E/flutter (20297): [ERROR:flutter/lib/ui/ui_dart_state.cc(171)] Unhandled Exception: NoSuchMethodError: The method 'toPosition' was called on null.
E/flutter (20297): Receiver: null
E/flutter (20297): Tried calling: toPosition()
E/flutter (20297): #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
E/flutter (20297): #1      PositionComponentEffect.initialize (package:flame/effects/effects.dart:78:25)
E/flutter (20297): #2      CombinedEffect.initialize (package:flame/effects/combined_effect.dart:28:11)
E/flutter (20297): #3      CombinedEffect.reset (package:flame/effects/combined_effect.dart:62:5)
E/flutter (20297): #4      SequenceEffect.initialize.<anonymous closure> (package:flame/effects/sequence_effect.dart:33:14)
E/flutter (20297): #5      List.forEach (dart:core-patch/growable_array.dart:313:8)
E/flutter (20297): #6      SequenceEffect.initialize (package:flame/effects/sequence_effect.dart:32:13)
E/flutter (20297): #7      PositionComponent.addEffect (package:flame/components/component.dart:156:26)
```

I think the cause of this issue is quite obvious. The CombinedEffect calls initialize from the reset function, but the component parameter is null because the CombinedEffect was not initialized before with a valid component.

## Development enviroment

Flame version: 0.26.0

Flutter doctor:

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel beta, 1.21.0-9.2.pre, on Microsoft Windows [Version 10.0.18362.1082], locale de-DE)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Chrome - develop for the web
[√] Android Studio (version 4.0)
[!] IntelliJ IDEA Community Edition (version 2019.2)
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
[√] Connected device (3 available)

! Doctor found issues in 1 category.
```

## Runtime

This issue is related to running on which platform?

 - [X] Android :android:
 - [X] iOS :apple:
 - [X] Web :browser:

I only tested this on Android and web, but I think this applies to all platforms.

## Minimal reproducible code

```dart
import 'dart:ui';

import 'package:flame/components/component.dart';
import 'package:flame/effects/combined_effect.dart';
import 'package:flame/effects/effects.dart';
import 'package:flame/game.dart';
import 'package:flutter/widgets.dart';

void main() {
  runApp(MyGame().widget);
}

class MyGame extends BaseGame {

  MyGame() {
    add(MyComponent());
  }

}

class MyComponent extends PositionComponent {

  MyComponent() {
    addEffect(SequenceEffect(effects: [
      CombinedEffect(effects: []),
    ]));
  }

  @override
  void render(Canvas c) {}

}
```

**Comment by spydon** [comment]
This is a bug indeed, good find!
If you have a solution we are always open for a PR.

**Comment by spydon** [comment]
Fixed in v1.0.0 (not yet released).
